src/makeCalculator.js

```diff
@@ -4,7 +4,37 @@
  * @return {object}
  */
 function makeCalculator() {
-  // write code here
+  let result = 0;
+
+  const calculator = {
+    add(num) {
+      result += num;
+    },
+    subtract(num) {
+      result -= num;
+    },
+    multiply(num) {
+      result *= num;
+    },
+    divide(num) {
+      result /= num;
+    },
+    reset() {
+      result = 0;
+
+      return this;
+    },
+    operate(callback, num) {
+      callback(num);
+
+      return this;
+    },
+    get result() {
+      return result;
+    },
+  };
+
+  return calculator;
 }
```

Comment:
The function `makeCalculator` is correctly named according to the task description. However, within the `calculator` object methods, the 'function' keyword is used unnecessarily. Use the method shorthand syntax for cleaner and more modern code.